common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java

```diff
@@ -219,6 +220,65 @@ ICacheRecord save(@NonNull AccountRecord accountRecord,
         return result;
     }
 
+    /**
+     * @param accountRecord      The {@link AccountRecord} to store.
+     * @param idTokenRecord      The {@link IdTokenRecord} to store.
+     * @param accessTokenRecord  The {@link AccessTokenRecord} to store.
+     * @param refreshTokenRecord The {@link RefreshTokenRecord} to store.
+     * @return The {@link ICacheRecord} result of this save action.
+     * @throws ClientException If the supplied Accounts or Credentials are schema invalid.
+     * @see OAuth2TokenCache#save(AccountRecord, IdTokenRecord)
+     */
+    ICacheRecord save(final @NonNull AccountRecord accountRecord,
+                      final @NonNull IdTokenRecord idTokenRecord,
+                      final @NonNull AccessTokenRecord accessTokenRecord,
+                      final @NonNull RefreshTokenRecord refreshTokenRecord) throws ClientException {
```

Comment:
@rpdome Can we make the RT `@Nullable` in this API and have the older API call this one? At a glance, it looks like that would allow for removal of some duplicated code

Reply:
I did that and reverted back - not liking the idea of having refreshToken nullable. It shouldn't be.

Reply:
OK - thanks @rpdome